topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.md

```diff
@@ -0,0 +1,471 @@
+>  * The pipeline was built with a different version of CellProfiler. The Galaxy tool currently uses CellProfiler 3.9.  
+>  * Modules used by the pipeline aren't available in Galaxy.  
+>  * Parameters for some CellProfiler modules are limited/constrained compared to the stand-alone version, most notably:  
+>    - Parameters require manual input from the user whereas, in the stand-alone version, some modules can inherit parameter values from other modules.
+>    - Input and output file locations are set by Galaxy and can't be set by the user.
+>    - Metadata extraction from file names is limited to a set of fixed patterns. 
+>
+> It is recommended to build a CellProfiler pipeline using the Galaxy interface if the pipeline is to be run by Galaxy.
+{: .warning}
+
+> ### Agenda
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+
+# Get data
+
+This tutorial will use a time-lapse recording of nuclei progressing through mitotic anaphase during early _Drosophila_ embryogenesis. The nuclei are labelled on chromatin with a GFP-histone marker and have been imaged every 7 seconds using a laser scanning confocal microscope with a 40X objective.
+The images are saved as a zip archive on Zenodo and need to be uploaded to the Galaxy server before they can be used.
+
+![Dividing nuclei](../../images/object-tracking-using-cell-profiler/Dividing_nuclei.gif "Time lapse recording of anaphase nuclei in a Drosophila embryo.")
+
+> ### {% icon hands_on %} Hands-on: Data upload
+>
+> 1. Create a new history for this tutorial.  
+>    When you log in for the first time, an empty, unnamed history is created by default. You can simply rename it.
+>    {% snippet snippets/create_new_history.md %}
+>
+> 2. Import {% icon galaxy-upload %} the files from [Zenodo]({{ page.zenodo_link }}) or from
+>    the shared data library.
+>    - **Important:** If setting the type to 'Auto-detect', make sure that after upload, the datatype is set to zip.
+>
+>    ```
+>    https://zenodo.org/api/files/e5d1bd5c-60a0-42e4-8f0d-a2ebc863c5d9/drosophila_sample.zip
+>    ```
+>
+>    {% snippet snippets/import_via_link.md %}
+>
+> 3. Rename {% icon galaxy-pencil %} the file to drosophila_embryo.zip
+>
+{: .hands_on}
+
+
+# Create a CellProfiler pipeline in Galaxy
+
+In this section, we will build a CellProfiler pipeline from scratch in Galaxy.
+We need to:  
+  - Read the images and the metadata  
+  - Convert the colour images to grayscale  
+  - Segment the nuclei  
+  - Extract features from the segmented nuclei  
+  - Perform tracking  
+  - Produce some useful output files  
+
+A pipeline is built by chaining together Galaxy tools representing CellProfiler modules and must start with the {% tool [Starting modules](toolshed.g2.bx.psu.edu/repos/bgruening/cp_common/cp_common/3.1.9+galaxy1) %} tool and end with the {% tool [CellProfiler](toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0) %}  tool.  
```

Comment:
Not strictly necessary to use the referencing syntax here. `{% tool Starting Modules %}` is fine in the future